cli/cluster.go

```diff
@@ -138,22 +144,42 @@ func runClusterAdd(args *docopt.Args) error {
 		return errors.New(fmt.Sprintf("Cluster %q does not exist and cannot be set as default.", s.Name))
 	}
 
+	var caPath string
+	if !args.Bool["--no-git"] && !args.Bool["--no-docker"] {
+		client, err := s.Client()
+		if err != nil {
+			return err
+		}
+		caPath, err = writeCACert(client, s.Name)
+		if err != nil {
+			return fmt.Errorf("Error writing CA certificate: %s", err)
+		}
+	}
+
 	if !args.Bool["--no-git"] {
 		if _, err := exec.LookPath("git"); err != nil {
 			if serr, ok := err.(*exec.Error); ok && serr.Err == exec.ErrNotFound {
 				return errors.New("Executable 'git' was not found. Use --no-git to skip git configuration")
 			}
 			return err
 		}
-		client, err := s.Client()
-		if err != nil {
+		if err := cfg.WriteGlobalGitConfig(s.GitURL, caPath); err != nil {
 			return err
 		}
-		caPath, err := writeCACert(client, s.Name)
+	}
+
+	if !args.Bool["--no-docker"] {
+		host, err := s.DockerHost()
 		if err != nil {
-			return fmt.Errorf("Error writing CA certificate: %s", err)
+			return err
 		}
-		if err := cfg.WriteGlobalGitConfig(s.GitURL, caPath); err != nil {
+		if err := dockerLogin(host, s.Key); err != nil {
+			if e, ok := err.(*exec.Error); ok && e.Err == exec.ErrNotFound {
+				err = errors.New("Executable 'docker' was not found. Use --no-docker to skip docker configuration")
```

Comment:
I decided to treat this the same as configuring git: unless you specify `--no-docker`, an error will be returned if Docker configuration fails.

Do you think we should default to configuring Docker but ignore configuration errors (just print a warning and move on), or perhaps switch it so that Docker configuration is only done if `--docker-url` (or `--docker`) is specified?